yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -68,8 +68,6 @@ private[spark] class YarnClientSchedulerBackend(
     // List of (target Client argument, environment variable, Spark property)
     val optionTuples =
       List(
-        ("--driver-memory", "SPARK_MASTER_MEMORY", "spark.master.memory"),
-        ("--driver-memory", "SPARK_DRIVER_MEMORY", "spark.driver.memory"),
```

Comment:
Since in `Client.scala` the `--driver-memory` passed by spark-submit is not used anymore.
I thought we've disscussed it before.

>@andrewor14 Ok, I got what you mean. I think I have a misunderstanding before. 
To solve this problem, should we just delete

("--driver-memory", "SPARK_MASTER_MEMORY", "spark.master.memory"),
("--driver-memory", "SPARK_DRIVER_MEMORY", "spark.driver.memory"), .
in YarnClientSchedulerBackend.scala?

>@WangTaoTheTonic that would fix it, but I think in addition to that we should also add a check in ClientArguments itself in case the user calls into the Client main class and specifying --driver-memory manually.